include/pagmo/problems/cec2009.hpp

```diff
@@ -0,0 +1,873 @@
+/* Copyright 2017 PaGMO development team
+
+This file is part of the PaGMO library.
+
+The PaGMO library is free software; you can redistribute it and/or modify
+it under the terms of either:
+
+  * the GNU Lesser General Public License as published by the Free
+    Software Foundation; either version 3 of the License, or (at your
+    option) any later version.
+
+or
+
+  * the GNU General Public License as published by the Free Software
+    Foundation; either version 3 of the License, or (at your option) any
+    later version.
+
+or both in parallel, as here.
+
+The PaGMO library is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received codetail::pi()es of the GNU General Public License and the
+GNU Lesser General Public License along with the PaGMO library.  If not,
+see https://www.gnu.org/licenses/. */
+
+#ifndef PAGMO_PROBLEM_CEC2009_HPP
+#define PAGMO_PROBLEM_CEC2009_HPP
+
+#include <cassert>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "../detail/constants.hpp"
+#include "../exceptions.hpp"
+#include "../problem.hpp" // needed for cereal registration macro
+#include "../types.hpp"
+
+#define MYSIGN(x) ((x) > 0 ? 1.0 : -1.0)
```

Comment:
* if we include some 3rd party headers which have ``MYSIGN`` defined, we gonna run into issues
* the name of the macro is too short and too common, the risk of name collision is high
* it's just a function, there's not a single reason why this should not be an inline function (if it needed to access information available only to macros, e.g., line number or file name, it would be ok, but it doesn't)